packages/core/useUrlSearchParams/index.ts

```diff
@@ -0,0 +1,81 @@
+import { computed, reactive, Ref, ref, watch } from 'vue-demi'
+import { useEventListener } from '../useEventListener'
+import { ConfigurableWindow, defaultWindow } from '../_configurable'
+
+export type UrlParams = {[key: string]: string[] | string}
+
+/**
+ * Reactive URLSearchParams
+ *
+ * @see   {@link https://vueuse.js.org/useUrlSearchParams}
+ * @param mode
+ * @param options
+ */
+export function useUrlSearchParams(
```

Comment:
Here we can have a generic for users to customize their expected type. You can leave this to me if you are not sure about how to do it.

Reply:
I've tried to change the composable to have a generic , but I failed resolving a TS error for accessing the object values by string index keys. Maybe you can help me there, i've tried several solutions, but none of them worked.

Reply:
I managed to fix the TS error by creating a `writeToPramsMap` function. Are you fine with 
```javascript
useUrlSearchParams<T extends Record<string, any> = UrlParams>
```